pkg/querysharding/analyzer_test.go

```diff
@@ -229,9 +242,9 @@ http_requests_total`,
 			shardingLabels: []string{"cluster", "pod", model.MetricNameLabel},
 		},
 		{
-			name:           "histogram quantile",
+			name:           "histogram quantile without",
 			expression:     "histogram_quantile(0.95, sum(rate(metric[1m])) without (le, cluster))",
-			shardingLabels: []string{"cluster"},
+			shardingLabels: []string{"cluster", "le"},
```

Comment:
This changes the current behavior. And I am not 100% sure whether this is correct or not.

Reply:
that query itself looks pretty weird since we have no bucket label left after the sum

Reply:
I think it should be shardable by cluster label; yeah. From the engine code:
```
			// This means that it has an invalid `le` label.
			if outputSeries == nil || !outputSeries.hasBucketValue {
				continue
			}

```
We will just skip all series with invalid bucket label and result will be empty. I guess if all are empty we could emit a warning diagnostic probably in the engine.

Reply:
Looks like we already have this test case. https://github.com/thanos-io/thanos/blob/main/pkg/querysharding/analyzer_test.go#L132

Then is it good to go?

Reply:
I think so but i always feel a bit uncertain when it comes to query sharding. But it seems ok to shard by bucket label iff there is no histogram_quantile function present in principle!